native/src/main/java/org/ballerinax/asb/util/ASBUtils.java

```diff
@@ -235,6 +246,499 @@ public static RecordType getRecordType(BTypedesc bTypedesc) {
+            subscriptionProperties.setBatchedOperationsEnabled(
+                    subscriptionConfig.getBooleanValue(
+                            ASBConstants.SUBSCRIPTION_RECORD_FIELD_ENABLE_BATCHED_OPERATIONS));
+        }
+        if (subscriptionConfig.containsKey(
+                ASBConstants.SUBSCRIPTION_RECORD_FIELD_FORWARD_DEAD_LETTERED_MESSAGES_TO)) {
+            subscriptionProperties.setForwardDeadLetteredMessagesTo(
+                    subscriptionConfig.getStringValue(
+                            ASBConstants.SUBSCRIPTION_RECORD_FIELD_FORWARD_DEAD_LETTERED_MESSAGES_TO).getValue());
+        }
+        if (subscriptionConfig.containsKey(ASBConstants.SUBSCRIPTION_RECORD_FIELD_FORWARD_TO)) {
+            subscriptionProperties.setForwardTo(
+                    subscriptionConfig.getStringValue(
+                            ASBConstants.SUBSCRIPTION_RECORD_FIELD_FORWARD_TO).getValue());
+        }
+        if (subscriptionConfig.containsKey(ASBConstants.SUBSCRIPTION_RECORD_FIELD_LOCK_DURATION)) {
+            subscriptionProperties.setLockDuration(
+                    getDurationFromBObject(subscriptionConfig.getMapValue(
+                            ASBConstants.SUBSCRIPTION_RECORD_FIELD_LOCK_DURATION)));
+        }
+        if (subscriptionConfig.containsKey(ASBConstants.SUBSCRIPTION_RECORD_FIELD_MAX_DELIVERY_COUNT)) {
+            subscriptionProperties.setMaxDeliveryCount(
+                    subscriptionConfig.getIntValue(
+                            ASBConstants.SUBSCRIPTION_RECORD_FIELD_MAX_DELIVERY_COUNT).intValue());
+        }
+        if (subscriptionConfig.containsKey(ASBConstants.SUBSCRIPTION_RECORD_FIELD_STATUS)) {
+            subscriptionProperties.setStatus(EntityStatus.fromString(
+                    subscriptionConfig.getStringValue(
+                            ASBConstants.SUBSCRIPTION_RECORD_FIELD_STATUS).getValue()));
+        }
+        if (subscriptionConfig.containsKey(ASBConstants.SUBSCRIPTION_RECORD_FIELD_USER_METADATA)) {
+            subscriptionProperties.setUserMetadata(
+                    subscriptionConfig.getStringValue(
+                            ASBConstants.SUBSCRIPTION_RECORD_FIELD_USER_METADATA).getValue());
+        }
+        return subscriptionProperties;
+    }
+
+    /**
+     * Get UpdatedRuleProperties from BMap.
+     *
+     * @param ruleConfig BMap containing the queue configurations.
+     * @return RuleProperties object.
+     */
+    public static RuleProperties getUpdatedRulePropertiesFromBObject(BMap<BString, Object> ruleConfig,
+                                                                     RuleProperties ruleProp) {
+        if (ruleConfig.containsKey(ASBConstants.RECORD_FIELD_ACTION)) {
+            ruleProp.setAction(new SqlRuleAction(
+                    ruleConfig.getStringValue(ASBConstants.RECORD_FIELD_ACTION).getValue()));
+        }
+        if (ruleConfig.containsKey(ASBConstants.RECORD_FIELD_ACTION)) {
```

Comment:
```suggestion
        if (ruleConfig.containsKey(ASBConstants.RECORD_FIELD_FILTER)) {
```

Reply:
Addressed in [9882b4a](https://github.com/ballerina-platform/module-ballerinax-azure-service-bus/pull/180/commits/9882b4a8780e73f7b894e0df833a9acf7abeca4d).